### PR TITLE
Add CI to release nightly free-threaded wheels

### PR DESCRIPTION
**Pull request by andfoy**
## Description
See #7464

This PR serves as a followup to #7463, adding a separate workflow to produce, test and upload nightly wheels to the Scientific Python Anaconda channel. 

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpydoc/blob/main/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
...
```

### CODE REVIEW

**Comment by andfoy** [comment]
I'm currently investigating an error in the test suite that is occurring under the latest 3.13 beta

**Comment by andfoy** [comment]
Wheels are being built successfully in my fork: https://github.com/andfoy/scikit-image/actions/runs/10287231013/job/28469781344

File: skimage/segmentation/tests/test_clear_border.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def test_clear_border_non_binary_out():
     image = np.array(
         [[1, 2, 3, 1, 2], [3, 3, 5, 4, 2], [3, 4, 5, 4, 2], [3, 3, 2, 1, 2]]
     )
-    out = np.empty_like(image)
+    out = image.copy()
```
Comment (andfoy) [inline skimage/segmentation/tests/test_clear_border.py:150]: This seems to be a legit issue in this function. `out` needs to be initialized as a copy of `image`, for some reason `empty_like` returned a previous chunk of memory which had similar values to `image`, thus it was passing before the memory allocator changes in CPython 3.13

Comment (lagru) [reply skimage/segmentation/tests/test_clear_border.py:150]: Uff, that's a good catch. Good reminder to never rely on the content of `np.empty`.

**Review by lagru — APPROVED** [review]
Looks good. Thanks @andfoy! :blush: 

What do you think about merging `build_linux_39_and_above_wheels` and `build_linux_free_threaded_wheels`? Would that be possible to make the extra `CIBW_FREE_THREADED_SUPPORT` and the like conditional on `cp313t-*` in the matrix? Maybe at a later point when we can remove the nightly installation?

**Comment by andfoy** [comment]
@lagru, thanks for the review, changes should be now up in 1d7c879, which is working in my fork: https://github.com/andfoy/scikit-image/actions/runs/10354531538/job/28660020992

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
Thank you @andfoy

File: .github/workflows/wheels_recipe.yml [hunk 2]
```diff
@@ -49,6 +49,17 @@ jobs:
...
+        if: endsWith(matrix.cibw_python, 't-*')
+        run: |
+          ANACONDA_URL="https://pypi.anaconda.org/scientific-python-nightly-wheels/simple"
+          NIGHTLY_INSTALL="pip install --pre -i $ANACONDA_URL cython numpy scipy pywavelets"
+          TEST_COMMAND="PYTHON_GIL=0 pytest --pyargs skimage"
+          echo "CIBW_PRERELEASE_PYTHONS=1" >> "$GITHUB_ENV"
+          echo "CIBW_FREE_THREADED_SUPPORT=1" >> "$GITHUB_ENV"
+          echo "CIBW_BEFORE_BUILD=$NIGHTLY_INSTALL" >> "$GITHUB_ENV"
+          echo "CIBW_BEFORE_TEST=$NIGHTLY_INSTALL" >> "$GITHUB_ENV"
+          echo "CIBW_TEST_COMMAND=$TEST_COMMAND" >> "$GITHUB_ENV"
```
Comment (lagru) [inline .github/workflows/wheels_recipe.yml:62]: Wouldn't have known how to do this! Thanks for figuring it out @andfoy. :blush:

**Pull request merged**